src/Common/GuardedPoolAllocatorState.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <base/defines.h>
+#include <cstddef>
+#include <cstdint>
+#include <cassert>
+
+#include <cstdio>
+
+namespace gwp_asan
```

Comment:
Note, that name of this namespaces overlaps with [GWP-ASan from llvm](https://llvm.org/docs/GwpAsan.html), let's use something different, like `clickhouse_gwp_asan` or `ch_gwp_asan`

Reply:
Will do in next update.